librtt/Display/Rtt_SpriteObject.cpp

```diff
@@ -302,6 +354,26 @@ SpriteObjectSequence::GetFrame( int index ) const
 	return ( IsConsecutiveFrames() ? (fStart+index) : fFrames[index] );
 }
 
+int
+SpriteObjectSequence::GetTimeForFrame( int frameIndex ) const
+{
+	if (GetTimeArray() == NULL)
+	{
+		return frameIndex * GetTimePerFrame();
+	}
+	else
+	{
+		Real summedTime = 0;
+		Real *timeArray = GetTimeArray();
+		for (int i = 0; i < frameIndex; ++i)
+		{
+			summedTime += timeArray[i];
```

Comment:
It seems like we never actually use `timeArray[i]` directly. We always use time of array start. So may be instead of storing `timeArray[i]` we can summarize it right away. Like, instead have a `frameEndTimeArray` where `frameEndTimeArray[i] = frameEndTimeArray[i-1] + timeArray[i]`, or something.

This way it seems we can skip ever iterating the `timeArray` or store cached `fTimeArrayCached*` values. I'm not sure, but it seems we can skip storing `timeArray` at all this way.

Reply:
I'm not a fan of not storing timeArray at all, since doing that would make SpriteObject::GetFrameIndexForDeltaTime more complex (and slightly less performant).

We could store an additional frameEndTimeArray or a cache which would remember the expected result for SpriteObjectSequence::GetTimeForFrame. But it sounds like premature optimising to me. Especially since SpriteObjectSequence::GetTimeForFrame is only executed when calling sprite:setFrame(), which I think would not be used a lot on sprites with variable frame times.